blues_refactor/example.py

```diff
@@ -0,0 +1,85 @@
+"""
+example.py: Provides an example script to run BLUES and
+benchmark the run on a given platform
+
+Authors: Samuel C. Gill
+Contributors: Nathan M. Lim, David L. Mobley
+
+* Benchmarking related code adapted from:
+https://github.com/pandegroup/openmm/blob/master/examples/benchmark.py
+(Authors: Peter Eastmen)
+
+version: 0.0.2 (WIP-Refactor)
+"""
+from __future__ import print_function
+from simtk import unit, openmm
+from simtk.openmm import app
+from alchemy import AbsoluteAlchemicalFactory, AlchemicalState
+
+import blues.utils as utils
+
+import blues.ncmc_switching as ncmc_switching
+from blues.smartdart import SmartDarting
+
+import blues_refactor.ncmc as ncmc
+from blues_refactor.simulationfactory import SimulationFactory
+
+import sys, os, parmed
+import numpy as np
+import mdtraj as md
+from mdtraj.reporters import HDF5Reporter
+from optparse import OptionParser
+
+def runNCMC(platform_name):
+    #Define some options
+    opt = { 'temperature' : 300.0, 'friction' : 1, 'dt' : 0.002,
+            'numIter' : 10, 'nstepsNC' : 10, 'nstepsMD' : 50,
+            'nonbondedMethod' : 'PME', 'nonbondedCutoff': 10, 'constraints': 'HBonds',
+            'trajectory_interval' : 10, 'reporter_interval' : 10, 'platform' : platform_name,
+            'verbose' : True }
+
+    #Defines ncmc move eqns for lambda peturbation of sterics/electrostatics
+    opt['functions'] = { 'lambda_sterics' : 'step(0.199999-lambda) + step(lambda-0.2)*step(0.8-lambda)*abs(lambda-0.5)*1/0.3 + step(lambda-0.800001)',
+                'lambda_electrostatics' : 'step(0.2-lambda)- 1/0.2*lambda*step(0.2-lambda) + 1/0.2*(lambda-0.8)*step(lambda-0.8)' }
+
+    # Obtain topologies/positions
+    prmtop = utils.get_data_filename('blues', 'tests/data/eqToluene.prmtop')
+    inpcrd = utils.get_data_filename('blues', 'tests/data/eqToluene.inpcrd')
+    struct = parmed.load_file(prmtop, xyz=inpcrd)
+    atom_indices = utils.atomIndexfromTop('LIG', struct.topology)
+
+    # Generate the MD, NCMC, ALCHEMICAL Simulation objects
+    sims = SimulationFactory(struct, atom_indices)
+    sims.createSimulationSet(opt)
+
+    # Calculate particle masses of object to be moved
+    from blues.modeller import LigandModeller
```

Comment:
@sgill2 - suggestions? What is the general purpose of this thing? `ObjectProperties`? `MoveProperties`? `MoveObject`? `MovingObject`? The last one seems to appeal to me most, as it might or might not be a ligand, and I also don't like the `modeller` terminology because of (a) UCSF Modeller, and (b) `Modeller` within OpenMM, as you note, @nathanmlim .